analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/CalculationsShouldNotOverflow.cs

```diff
@@ -40,29 +40,64 @@ public override bool ShouldExecute()
         }
     }
 
-    private sealed class SyntaxKindWalker : SafeCSharpSyntaxWalker
+    internal sealed class SyntaxKindWalker : SafeCSharpSyntaxWalker
     {
         public bool IsUnchecked { get; private set; }
         public bool HasOverflow { get; private set; }
 
         public override void Visit(SyntaxNode node)
         {
-            if (!IsUnchecked && !HasOverflow)
+            if (IsUnchecked)
             {
-                IsUnchecked = node.IsAnyKind(SyntaxKind.UncheckedStatement, SyntaxKind.UncheckedExpression);
-                HasOverflow = node.IsAnyKind(
-                    SyntaxKind.AddExpression,
-                    SyntaxKind.AddAssignmentExpression,
-                    SyntaxKind.MultiplyExpression,
-                    SyntaxKind.MultiplyAssignmentExpression,
-                    SyntaxKind.SubtractExpression,
-                    SyntaxKind.SubtractAssignmentExpression,
-                    SyntaxKind.PostDecrementExpression,
-                    SyntaxKind.PostIncrementExpression,
-                    SyntaxKind.PreDecrementExpression,
-                    SyntaxKind.PreIncrementExpression);
-                base.Visit(node);
+                return; // We have an unchecked context: stop visiting
             }
+            base.Visit(node);
+        }
+
+        public override void VisitBinaryExpression(BinaryExpressionSyntax node)
+        {
+            HasOverflow |= node.Kind() is
+                SyntaxKind.AddExpression or
+                SyntaxKind.MultiplyExpression or
+                SyntaxKind.SubtractExpression;
```

Comment:
Can you put these on a single line? Othwerwise, the `or` would need to be on the beginning of next line, similar to `||`